Mathlib/Data/Matrix/ColumnRowPartitioned.lean

```diff
@@ -0,0 +1,229 @@
+/-
+Copyright (c) 2023 Mohanad ahmed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed
+-/
+
+import Mathlib.Data.Matrix.Basic
+import Mathlib.Data.Matrix.Block
+import Mathlib.LinearAlgebra.Matrix.NonsingularInverse
+
+/-! # Block Matrices from Rows and Columns
+
+This file provides the basic definitions of matrices composed from columns and rows.
+The concatenation of two matrices with the same row indices can be expressed as
+`A = fromColumns A₁ A₂` the concatenation of two matrices with the same column indices
+can be expressed as `B = fromRows B₁ B₂`.
+
+We then provide a few lemmas that deal with the products of these with each other and
+with block matrices
+
+## Tags
+column matrices, row matrices, column row block matrices
+-/
+
+namespace Matrix
+
+variable {R: Type _}
+variable {M M₁ M₂ N N₁ N₂: Type _}
+variable [Fintype M][Fintype M₁][Fintype M₂]
+variable [Fintype N][Fintype N₁][Fintype N₂]
+variable [DecidableEq M][DecidableEq M₁][DecidableEq M₂]
+variable [DecidableEq N][DecidableEq N₁][DecidableEq N₂]
+
+/-- Concatenate together two matrices A₁[M₁ × N] and A₂[M₂ × N] with the same columns (N) to get a
+bigger matrix indexed by [(M₁ ⊕ M₂) × N] -/
```

Comment:
I'm not familiar with the `A₁[M₁ × N]` notation, maybe you have a reference for it?